src/sorted_vector.jl

```diff
@@ -0,0 +1,52 @@
+"""
+A `SortedVector` behaves like a standard Julia `Vector`, except that its elements are stored in sorted order, with an optional function `by` that determines the sorting order in the same way as `Base.searchsorted`.
+"""
+immutable SortedVector{T, F<:Function}
+    data::Vector{T}
```

Comment:
AFAICT you only use functions in the `AbstractArray` interface, right? Anyway, even if a method fails for some peculiar types, it cannot be worse than if you only accept `Vector`.